flow/raster_cache.cc

```diff
@@ -126,7 +126,10 @@ static std::unique_ptr<RasterCacheResult> Rasterize(
     const std::function<void(SkCanvas*)>& draw_function) {
   TRACE_EVENT0("flutter", "RasterCachePopulate");
 
-  SkRect cache_rect = RasterCache::GetDeviceBounds(logical_rect, ctm);
+  SkRect dest_rect = RasterCache::GetDeviceBounds(logical_rect, ctm);
+  // we always round out here so that the texture is integer sized.
+  SkIRect cache_rect;
+  dest_rect.roundOut(&cache_rect);
```

Comment:
This can choose a cache image size that is 1 pixel larger than needed in the non-ICTM case because our translation will be by dest_rect.left/top which may move it closer to the origin of the cache image by enough to not need that last pixel on the right and bottom. A minor issue, but could be fixed by using the ceil of the dest_rect width and height instead of the height of the cache_rect.

With ICTM, dest_rect is all integers anyway so the ceil of its w/h are just the w/h.

Without ICTM, the w/h indicate the number of pixels from the left/topmost coordinate in the drawing which are mapped to 0,0 in the cache image by the translate below and so the ceil of the w/h represent the number of whole pixels which it will take up.